prince_cr/solvers/propagation.py

```diff
@@ -448,29 +452,34 @@ def eqn_deriv_mkl(self, z, state, *args):
             self.current_z_rates = z
 
         res = np.zeros(state.shape)
-        
         if matrix_input:
             state = np.ascontiguousarray(state)
             res = csrmm(1., self.jacobian, state, 0., res)
         else:
             res = csrmv(1., self.jacobian, state, 0., res)
-            
 
         if self.enable_injection_jacobian:
-            res += self.injection(1., z)[:, np.newaxis]
+            try:
+                res += self.injection(1., z)[:, np.newaxis]
+            except ValueError:
+                res += self.injection(1., z)
```

Comment:
This fixes the problem with the Euler solver, right?

This exception here is a little confusing. Maybe we should rather adjust this in the Euler solver class.

Or atleast leave a comment:
```suggestion
            # catch error due to different shape of res in UHECRPropagationSolverEULER
            try:
                res += self.injection(1., z)[:, np.newaxis]
            except ValueError:
                res += self.injection(1., z)
```